File: Python/src/Problem007.py
__author__ = 'Joel'


def is_prime(value):
    if value <= 1: return False
    for i in range(2, value // 2 + 1):
        if value % i == 0: return False
    return True


i = 0
val = 2
while True:
    if is_prime(val):
        i += 1
        if i == 10001:
            break
    val += 1
print(val)

File: Java/src/Problem007.java
public class Problem007 {
	public static void main(String[] args) {
		int i = 0;
		int val = 2;
		while (true) {
			if (isPrime(val) && ++i == 10001) {
				break;
			}
			val++;
		}
		System.out.println(val);
	}

	private static boolean isPrime(int value) {
		if (value <= 1) return false;
		for (int i = 2; i <= value / 2; i++) {
			if (value % i == 0) return false;
		}
		return true;
	}
}